addons/l10n_id/models/qris_transaction.py

```diff
@@ -0,0 +1,98 @@
+import datetime
+import logging
+import pprint
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+QRIS_FETCH_STATUS_URL = "https://qris.online/restapi/qris/checkpaid_qris.php"
+
+class QrisTransaction(models.Model):
+    _name = "l10n_id.qris.transaction"
+    _description = "Records of QRIS Transactions"
+
+    model = fields.Char(readonly=True)
+    res_id = fields.Integer(readonly=True)
+    partner_bank_id = fields.Many2one('res.partner.bank', string='Recipient Bank')
+
+    # QRIS related information
+    qris_content = fields.Char("QR Content")
+    qris_invoiceid = fields.Char("QRIS Transaction Number")
+    trx_date = fields.Datetime("Transaction Creation Time")
+    paid = fields.Boolean()
+    amount = fields.Float()
+
+    _sql_constraints = [('ref_uniq', "unique(model, res_id)",
+                         "There's already a transaction with the same reference")]
+
+    def _create_payment_vals(self, move, **extra_create_values):
+        # reference to _create_payment from payment.transaction
+        return {
+            'amount': self.amount,
+            'payment_type': 'inbound',
+            'currency_id': move.currency_id.id,
+            'partner_id': move.partner_id.commercial_partner_id.id,
+            'partner_type': 'customer',
+            'journal_id': self.partner_bank_id.journal_id.id,
+            'ref': move.payment_reference,
+            **extra_create_values
+        }
+
+    def _cron_fetch_payment_status(self):
+        """ Get all qris transaction that is within 30 minutes and fetch each status"""
+        prev_30_minutes = fields.Datetime.now() - datetime.timedelta(minutes=30)
```

Comment:
good point.
In this case, then I'll just pick up all unpaid transactions because those outdated unpaid transactions will be picked up by `_clean_transactions()`